code/modules/mob/living/carbon/carbon.dm

```diff
@@ -124,42 +123,32 @@ mob/living
 	shock_damage *= siemens_coeff
 	if(shock_damage<1 && !override)
 		return 0
-
-	src.apply_damage(shock_damage, BURN, def_zone, used_weapon="Electrocution")
-
-	if(heart_attack && prob(25))
-		heart_attack = 0
-	playsound(loc, "sparks", 50, 1, -1)
-	if (shock_damage < 10)
-		src.visible_message(
-			"\red [src] was mildly shocked by the [source].", \
-			"\red You feel a mild shock course through your body.", \
-			"\red You hear a light zapping." \
-		)
-		jitteriness += (rand(2,4))//mostly for the swarmer trap
-		do_jitter_animation(jitteriness)
-	if (shock_damage > 10)
-		if (shock_damage < 200)
-			src.visible_message(
-				"\red [src] was shocked by the [source]!", \
-				"\red <B>You feel a powerful shock course through your body!</B>", \
-				"\red You hear a heavy electrical crack." \
-			)
-		jitteriness += 1000 //High numbers for violent convulsions
-		do_jitter_animation(jitteriness)
-		stuttering += 2
+	if(reagents.has_reagent("teslium"))
+		shock_damage *= 1.5 //If the mob has teslium in their body, shocks are 50% more damaging!
+	take_overall_damage(0,shock_damage)
```

Comment:
take_overall_damage targets multiple areas---and the parameters for it are (brute,burn). The problem with the previous one is that it always targeted the hands---which gave artificially high protection to people with robotic limbs.

Also, just because you get shocked at the location of your hands (for not wearing gloves) doesn't mean that  that's necessarily what would be burnt; anyone who's deal with electricity will tell you how crazy it can be in that regard.